src/java.base/share/classes/java/lang/String.java

```diff
@@ -1008,6 +1008,26 @@ private static boolean isASCII(byte[] src) {
 
     //////////////////////////////// utf8 ////////////////////////////////////
 
+    /**
+     * Decodes ASCII from the source byte array into the destination
+     * char array.
+     *
+     * @return the number of bytes successfully decoded, at most len
+     */
+    /* package-private */
```

Comment:
Some more explanation would be helpful here, as it is accessed from System for shared secrets.

Reply:
Ok, added a short comment